test/framework/management.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ComponentGenerator is used to install components, generally any YAML bundle.
+type ComponentGenerator interface {
+	// GetName returns the name of the component.
+	GetName() string
+	// Components return the YAML bundle.
+	Components() ([]byte, error)
+}
+
+// Applier is an interface around applying YAML to a cluster
+type Applier interface {
+	// Apply allows us to apply YAML to the cluster, `kubectl apply`
+	Apply([]byte) error
+}
+
+// ManagementCluster are all the features we need out of a kubernetes cluster to qualify as a management cluster.
+type ManagementCluster interface {
+	Applier
+	// Wait allows us to wait for something in the cluster, `kubectl wait`
+	Wait(...string) error
+	// Teardown will completely clean up the ManagementCluster.
+	// Generally to be used in the AfterSuite function if a management cluster is shared between tests.
+	Teardown() error
+	// LoadImage is `kind` specific. It injects images into the management cluster and is how we test local code.
+	LoadImage(string) error
```

Comment:
Do we assume Kind?

Reply:
This doesn't strictly assume kind, but as you can tell this interface was generally designed around kind. But! I don't think this interface needs this function. Let me work on simplifying this interface.

Reply:
I'm fine requiring it for now